### PR TITLE
show file icons

### PR DESCRIPTION
**Pull request by kounoike**
### Before submitting a pull-request to GitBucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [ ] rebased my branch over master
- [x] verified that project is compiling
- [ ] verified that tests are passing
- [ ] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [ ] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

### CODE REVIEW

**Comment by aadrian** [comment]
@kounoike very nice improvement

It would be however even more useful if this were some sort of plug-in/theme, or integrate with this one too: https://github.com/gitbucket-plugins/gitbucket-explorer-plugin at least, so that file navigation to be uniform.

**Comment by kounoike** [comment]
This PR uses javascript for get icon type from filename. so, plugins can use same approach. I wrote that use icon in gitbucket-explorer-plugin.

image

**Comment by aadrian** [comment]
> This PR uses javascript for get icon type from filename. so, plugins can use same approach

@kounoike this is fantastic !

**Comment by takezoe** [comment]
I prefer current simple icons (just my opinion). Anyway, it would be nice if we can provide such presentation enhancement as plugin.

**Comment by lavvy** [comment]
Yeah . I think this should come as a plugin. I prefer the simple icons too,
as is now.

**Comment by kounoike** [comment]
Find file feature is implemented by javascript. So, it is hard to implement as plugin. How about to option in system settings?

**Comment by takezoe** [comment]
You can inject JavaScript from plugins using this extention point:
https://github.com/gitbucket/gitbucket/blob/ecc50cd2ae00f2456b140055076127b5ffa709a7/src/main/scala/gitbucket/core/plugin/Plugin.scala#L50-L59

But I think that maybe we should merge generating `data-filename` attribute. It makes easy to create such plugins.

File: src/main/twirl/gitbucket/core/repo/find.scala.html [hunk 1]
```diff
@@ -106,6 +106,7 @@
       for(var i = 0; i < p.length; i++){
         var row = template.clone();
         row.find("a").attr("href", pathBase + "/" + encodeURIComponent(p[i].string).replace(/%2F/g, '/')).html(string_score_highlight(p[i], '<b>'));
+        row.find("i").addClass(FileIcons.getClassWithColor(p[i].string));
```
[the remark this answers is not in the document]
Comment (takezoe) [reply src/main/twirl/gitbucket/core/repo/find.scala.html]: Hmm... I didn't aware this point.

**Comment by kounoike** [comment]
`file-icons-js` has one more option. File icon can draw without color.

image

So, how about adding an option? It has these options:

- simple
- file icon without color
- file icon with color

**Comment by aadrian** [comment]
>So, how about adding an option? It has these options:
>   - simple
>   - file icon without color
>   - file icon with color

+1 
that would be fantastic.

**Comment by zexxar** [comment]
Being able to easily select the icon style would be a great improvement IMO.

**Comment by kounoike** [comment]
I added support to select icon style.

**Comment by kounoike** [comment]
I wrote this feature as [gitbucket-fileicon-plugin](https://github.com/kounoike/gitbucket-fileicon-plugin).

**Pull request closed without merging**